pyproject.toml

```diff
@@ -114,6 +114,7 @@ filterwarnings = [
     'ignore::asdf.exceptions.AsdfDeprecationWarning:asdf.asdftypes',
     'ignore:numpy.ndarray size changed:RuntimeWarning',
     'ignore:.*from astropy.io.misc.asdf.*subclasses the deprecated CustomType.*:asdf.exceptions.AsdfDeprecationWarning',
+    'ignore::asdf.exceptions.AsdfDeprecationWarning:asdf.resolver:7'
```

Comment:
@WilliamJamieson Can I get your thoughts on how to handle this deprecation? Based on the previous PR having this line is not preferred which makes the approach taken in this PR (move `resolver.py`  to `_resolver.py` and add a stub in `resolver.py` that issues a warning on direct use) unacceptable.